opentelemetry-sdk/src/logs/error.rs

```diff
@@ -54,10 +56,32 @@ impl From<&'static str> for LogError {
 
 impl<T> From<PoisonError<T>> for LogError {
     fn from(err: PoisonError<T>) -> Self {
-        LogError::Other(err.to_string().into())
+        LogError::ClientFailed(err.to_string().into())
     }
 }
+
 /// Wrap type for string
 #[derive(Error, Debug)]
 #[error("{0}")]
 struct Custom(String);
+
+/// Errors returned during shutdown
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum ShutdownError {
+    /// The export client failed while holding the client lock. It is not
+    /// possible to complete the shutdown and a retry will not help.
+    /// This is something that should not happen and should likely emit some diagnostic.
+    #[error("export client failed while holding lock; cannot retry.")]
+    ClientFailed(String),
+
+    /// Other errors propagated from log SDK that weren't covered above.
+    #[error(transparent)]
+    Other(#[from] Box<dyn std::error::Error + Send + Sync + 'static>),
+}
+
+impl<T> From<PoisonError<T>> for ShutdownError {
+    fn from(err: PoisonError<T>) -> Self {
+        ShutdownError::ClientFailed(format!("Mutex poisoned during shutdown: {}", err))
```

Comment:
I think wrapping up the formatted inner error is plenty of context here given the narrow failure mode. I also considered wrapping up `Box<Err>` , but I think this is 1/ confusing in the interface and 2/ not at all actionable. 

There's some [good stuff on canonical's rust best practices doc](https://github.com/canonical/rust-best-practices/blob/main/best-practices.md#error-conversion) on this stuff